app/src/main/cpp/emu_jni.cpp

```diff
@@ -84,10 +89,39 @@ extern "C" JNIEXPORT void Java_emu_skyline_EmulationActivity_setSurface(JNIEnv *
     JniMtx.unlock();
 }
 
-extern "C" JNIEXPORT jint Java_emu_skyline_EmulationActivity_getFps(JNIEnv *env, jobject thiz) {
+extern "C" JNIEXPORT jint Java_emu_skyline_EmulationActivity_getFps(JNIEnv *, jobject) {
     return fps;
 }
 
-extern "C" JNIEXPORT jfloat Java_emu_skyline_EmulationActivity_getFrametime(JNIEnv *env, jobject thiz) {
+extern "C" JNIEXPORT jfloat Java_emu_skyline_EmulationActivity_getFrametime(JNIEnv *, jobject) {
     return static_cast<float>(frametime) / 100;
 }
+
+extern "C" JNIEXPORT void JNICALL Java_emu_skyline_EmulationActivity_setController(JNIEnv *, jobject, jint index, jint type, jint partnerIndex) {
+    while (input == nullptr)
+        asm("yield");
```

Comment:
I think there's little point of doing something except spinning because it realistically wastes more resources due to the extremely low contention over here and is mostly irrelevant to performance due to being at the start.

Reply:
It's still an ugly way to do it, see the solution i proposed in discord earlier, it's much cleaner and can be used in the future for other things

Reply:
I will add a comment according to the Markus x ByLaws pact.